src/shared/exec-util.c

```diff
@@ -365,8 +366,76 @@ static int gather_environment_consume(int fd, void *arg) {
         return r;
 }
 
+int exec_command_flags_from_strv(char **ex_opts, ExecCommandFlags *flags) {
+        ExecCommandFlags ex_flag = 0;
+        char **opt = NULL;
+
+        assert(flags);
+
+        STRV_FOREACH(opt, ex_opts) {
+                ex_flag = exec_command_flags_from_string(*opt);
+                if (ex_flag != _EXEC_COMMAND_FLAGS_INVALID)
+                        *flags |= ex_flag;
```

Comment:
this clobbers the return value of this call even if it fails later. We generally try to avoid that: return parameters should only be set on success, and on failure return parameters should be left untouched. Also see https://systemd.io/CODING_STYLE